app/models/job.rb

```diff
@@ -7,6 +7,9 @@ class Job < ActiveRecord::Base
   belongs_to :author, foreign_key: :user_id, class_name: "User"
 
   scope :published, -> { where(state: :published).order(created_at: :desc) }
+  scope :search, (lambda do |query|
```

Comment:
The "stabby lambda" form is [recommended](http://guides.rubyonrails.org/active_record_querying.html#scopes) for scopes. You can also avoid repetition using the [hash conditions](http://guides.rubyonrails.org/active_record_querying.html#hash-conditions) form.

```ruby
scope :search, -> (query) {
  where "title LIKE :pattern OR description LIKE :pattern",
    pattern: "%#{query}%"
}
```

The Query Interface guide is great at covering all of this. http://guides.rubyonrails.org/active_record_querying.html